examples/5_teach_robot_record_dataset.py

```diff
@@ -58,22 +51,34 @@ def main(task: str, backend: str, backend_api_key: str, disable_audio: bool) ->
         model_src=backend,
     )
 
-    # Prefer to use use_pyaudio=False for MAC.
-    audio = AudioAgent(use_pyaudio=False)
-
     # Use sim robot for this example.
     robot = SimRobot()
-    robot_recorder = RobotRecorder(robot, frequency_hz=5)
+    # Initialize the recorder for the robot.
+    robot.init_recorder(
```

Comment:
with robot.record():


Talk about whats going on under the hood in a docstring or a markdown doc